column_buffer.go

```diff
@@ -900,6 +900,95 @@ func (col *booleanColumnBuffer) ReadValuesAt(values []Value, offset int64) (n in
 	}
 }
 
+type int8ColumnBuffer struct{ int32Page }
```

Comment:
I chose to use an int32Page as the underlying structure for the int8ColumnBuffer because it minimized the code required to get the implementation working and because I wasn't sure whether it made sense to have a separate page for a logical type with the same underlying physical type. Happy for input here.